crates/bevy_egui/src/render.rs

```diff
@@ -0,0 +1,240 @@
+use bevy::{
+    prelude::*,
+    render::{
+        camera::{
+            ActiveCameras,
+            Camera,
+            OrthographicProjection,
+            VisibleEntities,
+            WindowOrigin
+        },
+        pass::{
+            LoadOp,
+            Operations,
+            PassDescriptor,
+            RenderPassDepthStencilAttachmentDescriptor,
+            TextureAttachment,
+        },
+        pipeline::*,
+        render_graph::{
+            base,
+            AssetRenderResourcesNode,
+            CameraNode,
+            NodeId,
+            PassNode,
+            RenderGraph,
+            WindowSwapChainNode,
+            WindowTextureNode,
+        },
+        shader::{
+            Shader,
+            ShaderStage,
+            ShaderStages,
+        },
+        texture::TextureFormat
+    }
+};
+
+use crate::egui_node::{
+    EguiNode,
+    EguiSystemNode
+};
+
+#[derive(Bundle)]
+pub struct EguiCameraComponents {
+    pub camera: Camera,
+    pub orthographic_projection: OrthographicProjection,
+    pub visible_entities: VisibleEntities,
+    pub transform: Transform,
+    pub global_transform: GlobalTransform,
+}
+
+impl Default for EguiCameraComponents {
+    fn default() -> Self {
+        // we want 0 to be "closest" and +far to be "farthest" in 2d, so we offset
+        // the camera's translation by far and use a right handed coordinate system
+        let far = 1000.0;
+        EguiCameraComponents {
+            camera: Camera {
+                name: Some(camera::EGUI_CAMERA_NAME.to_string()),
+                ..Default::default()
+            },
+            orthographic_projection: OrthographicProjection {
+                far,
+                window_origin: WindowOrigin::BottomLeft,
+                ..Default::default()
+            },
+            visible_entities: Default::default(),
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, far - 0.1)),
+            global_transform: Default::default(),
+        }
+    }
+}
+
+pub const EGUI_PIPELINE_HANDLE: Handle<PipelineDescriptor> =
+    // No clue where the hell this number comes from, but I changed a few of them to be slightly different
+    Handle::from_u128(323432602125399287835112542539754486265);
```

Comment:
😨 

At least explain what `EGUI_PIPELINE_HANDLE` is and what it's used for if you're going to be this evil

Reply:
To be clear, this is literally what bevy does everywhere... I at least put a comment about how weird it is! Definitely evil though...

Reply:
I can't find the definition of `from_u128` in Bevy, but I assume this is just a GUID of some kind? (i.e. the actual value is meaningless, as long as it's different to other values).

Reply:
It just makes a UUID from the u128, so yeah pretty much